### DISCUSSION TITLE
[UI improvements / Feature Suggestion] Expand/Collapse toggles for browsing filesystem and visualising hierarchy

### DISCUSSION THREAD

**Discussion by spiritually-soup** [mikekazakov/nimble-commander#362] [Ideas]
I saw the suggestion for a [sidebar panel](https://github.com/mikekazakov/nimble-commander/discussions/236) which I would personally find helpful.

But for a more (visually) lightweight and space-efficient layout, I'd love to see an option for **toggle functionality** like there is in Finder's List View.

**Reasoning:**
1. Tree-style views help better visualise the folder hierarchy and keeps context clues visible, which makes it clearer where the file is within your system (if you're using NC to move stuff between folders, which is probably a frequent use case).

2. Tree-style directory navigation makes it easier to find the thing if you can't remember the file name and don't know exactly where you put the thing. 

3. Enables you to browse >1 folder within the same pane, without doing it in a new tab which could get messy quickly

4. Unless I'm an idiot who hasn't found the setting yet, I think there isn't a 'back' button? (how do I go back?)
And the 'Go' menu is only a single layer deep, meaning it could take some time to navigate back through a few layers of folders.

<details><summary>PICS & SUCH</summary>
<p>

> Once I've reached my overall destination folder, I often find it easier to browse through my filesystem with toggles.
<img width="600" alt="Xnip2024-08-27_18-22-17" src="https://github.com/user-attachments/assets/1f5fa6c6-ae9c-4d44-aa61-4c6b9c6275f6">

.
> Limited context view + can't 'browse' backwards?
<img width="400" alt="Xnip2024-08-27_18-29-58" src="https://github.com/user-attachments/assets/24104332-4ec0-4616-8375-f1aebb17470a">

.
> Main menu button is only 1 layer deep, so you have to click through before navigating further.
<img width="400" alt="Xnip2024-08-27_18-13-52" src="https://github.com/user-attachments/assets/18b34102-a1a5-4062-81b3-03bcd0ca7e54">

> vs: multi-layer menu
<img width="300" alt="Xnip2024-08-27_18-23-38" src="https://github.com/user-attachments/assets/c08ad31b-7b9e-4883-a705-3723e5a95ee8">

(if a toggle function is not in the cards, implementing 1-2 more layers of menu  would be helpful I think!)

</p>
</details> 

Also, for your consideration -- a 'sort' button in the Toolbar (potentially activate-able by user, or there by default)? When the NC window is nowhere near your menubar and you're not a power keyboard user, it takes more effort to sort by fields that you haven't enabled into that view.

**Comment by mikekazakov** [comment]
Thank you for the suggestions. They are welcome, though community contributions would be required to bring them to life, as I'm already stretched too thin to take on additional features.  

To answer the questions:
- NC is designed to be controlled primarily by keyboard
- the cmd+[ and cmd+] hotkeys can be used to navigate back and forth
- sorting mode can be changed by clicking at the sorting indicator in each panel, e.g. "n"
